### PR TITLE
WILMA v1.0

### PR DESCRIPTION
**Pull request by nick-mcguffin**
---
name: WILMA v1.0
about: Merge Request
title: 'WILMA v1.0 PR'
labels: ''
assignees: 'Nicholas McGuffin', 'Nathan Snow'

---

**PR Type** *(Select all that apply)*
  - [x] Feature 
  - [ ] Bug fix
  - [x] Improvement 
  - [ ] Other

**Feature Description**

Replace this with a description of what your PR covers
You can use screenshots or examples if it helps your explanation

**Submission Checklist** *(do not remove)*

Use the following as a guide to ensure your PR is complete and acceptable:

- [ ] You are NOT attempting to merge into the `master` branch
- [ ] You are requesting to merge your branch into the `development` branch or one of its sub-branches
- [x] Your code compiles and runs without errors (on as many operating systems as practicable
- [x] Code has sufficient Javadoc comments (giving examples where needed)
- [x] Code is clear and readable, being sure to use meaningful names and effective whitespace
- [x] Appropriate [naming convention](https://en.wikipedia.org/wiki/Naming_convention_(programming)) is used according to the language used
- [x] Code duplication is minimized through use of polymorphism and inheritance
- [x] Dependencies have been checked for vulnerabilities
- [x] There is sufficient unit tests for the code and all tests pass
- [x] Unit tests utilize mock data/tools and don't write data to any databases
- [x] There is no logic in any views, use service/controller methods as helpers 
- [x] Use spaces between method parameters `[1, 2, 3]` not `[1,2,3]` and around operators `a + b` not `a+b`
- [x] All work has been thoroughly checked for grammatical and spelling errors
- [x] Documentation is to be clear, concise, and written in English
- [x] Changes are documented in a `README.md` if applicable

### CODE REVIEW

**Comment by nick-mcguffin** [comment]
One minor error - Partner Expression of Interest throws  errors: 
- Filter not working
- pendingPositions not defined.

File: web/src/main/java/com/wilma/web/controller/web/portal/StudentPortalController.java [hunk 1]
```diff
@@ -1,27 +1,144 @@
...
+                "repliesForPosts", forumService.getPostRepliesByCategory(category)));
+        return "/student/forum/forum-thread";
+    }
+    //endregion
+
+    //region Resume management
+    @GetMapping("/resume-management")
     public String resumeManagement(Model model) {
         model.addAllAttributes(Map.of(
                 "currentPage", "Resume Management",
```
Comment (nathansnow1981) [inline web/src/main/java/com/wilma/web/controller/web/portal/StudentPortalController.java:144]: Current page needs to be consistent with the element name to set the button "active"
```suggestion
"currentPage", "resume-management",
```

File: web/src/main/java/com/wilma/web/controller/web/portal/PartnerPortalController.java [hunk 2]
```diff
@@ -28,26 +46,146 @@ public String dashboard(Model model) {
...
+        return "/partner/forum/forum-thread";
+
+    }
+    //endregion
+
+    //region Expressions of interest
+    @GetMapping("/expressions-of-interest")
+    public String expressionsOfInterest(Model model) {
+        model.addAllAttributes(Map.of(
+                "currentPage", "Expressions Of Interest",
```
Comment (nathansnow1981) [inline web/src/main/java/com/wilma/web/controller/web/portal/PartnerPortalController.java:166]: Menu item not being selected
```suggestion
"currentPage", "expressions-of-interest",
```

File: web/src/main/java/com/wilma/web/controller/web/portal/PartnerPortalController.java [hunk 3]
```diff
@@ -28,26 +46,146 @@ public String dashboard(Model model) {
...
+        ));
+        return "/partner/expressions-of-interest";
+    }
+    //endregion
+
+    //region Profile
+    @GetMapping("/profile")
+    public String partnerProfile(Model model) {
+        model.addAllAttributes(Map.of(
+                "currentPage", "Profile",
```
Comment (nathansnow1981) [inline web/src/main/java/com/wilma/web/controller/web/portal/PartnerPortalController.java:184]: Menu item not being selected
```suggestion
"currentPage", "profile",
```

File: web/src/main/java/com/wilma/web/controller/web/portal/EducatorPortalController.java [hunk 4]
```diff
@@ -29,31 +46,147 @@ public String dashboard(Model model) {
...
+                "repliesForPosts", forumService.getPostRepliesByCategory(category)));
+        return new RedirectView("/educator/forum-thread?category=" + category);
+    }
+    //endregion
+
+    //region Expressions of interest
+    @GetMapping("/expressions-of-interest")
+    public String expressionsOfInterest(Model model) {
+        model.addAllAttributes(Map.of(
+                "currentPage", "Expressions Of Interest",
```
Comment (nathansnow1981) [inline web/src/main/java/com/wilma/web/controller/web/portal/EducatorPortalController.java:162]: Menu item not being selected
```suggestion
"currentPage", "expressions-of-interest",
```

File: web/src/main/java/com/wilma/web/controller/web/portal/EducatorPortalController.java [hunk 5]
```diff
@@ -29,31 +46,147 @@ public String dashboard(Model model) {
...
+        ));
+        return "/educator/expressions-of-interest";
+    }
+    //endregion
+
+    //region Profile
+    @GetMapping("/profile")
+    public String EducatorProfile(Model model) {
+        model.addAllAttributes(Map.of(
+                "currentPage", "Profile",
```
Comment (nathansnow1981) [inline web/src/main/java/com/wilma/web/controller/web/portal/EducatorPortalController.java:185]: Menu item not being selected
```suggestion
"currentPage", "profile",
```

**Review by nathansnow1981 — COMMENTED** [review]
Found another couple of small things where the sidebar menu buttons are not being highlighted (set to active) due to naming inconsistencies.
Not a big deal by any means, but just noticed them so wanted to document it

File: web/src/main/java/com/wilma/web/controller/web/portal/StudentPortalController.java (hunk 1 above)
Comment (nick-mcguffin) [reply web/src/main/java/com/wilma/web/controller/web/portal/StudentPortalController.java:144]: Awesome, great find.

**Review by nathansnow1981 — APPROVED** [review]
Looking slick 👌

**Pull request merged**